### PR TITLE
Add first draft of week 3 git slides

### PR DESCRIPTION
**Pull request by jgkamat**
Still a WIP, but feedback is appreciated.

### CODE REVIEW

**Comment by joshhting** [comment]
Can you change it to merge from `RoboJackets:weekThreeGitSlides` instead so I can push to it? Or do we have to do it from a new pr?

**Comment by jgkamat** [comment]
@joshhting you can push to it, I have 'allow edits from maintainers' checked. If you have push to this repo you have push to that branch on  my fork.

```
git remote add jgkamat git@github.com:jgkamat/software-training.git
git fetch jgkamat
git checkout weekThreeGitSlides
git pull jgkamat weekThreeGitSlides
git push jgkamat weekThreeGitSlides
```

I can switch it over if you really want that though (it would need a new PR).

I would also reccomend hub https://github.com/github/hub (which lets you do `git fetch jgkamat` and that's it).

**Comment by joshhting** [comment]
Also the students are going to be practicing these commands on the software training repo right?

**Comment by jgkamat** [comment]
I tried to structure it so it dosen't matter, they can do it on a bare git repo or on the software training repo. I never touch master, so their actual state shouldn't be affected by this demo.

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org:48]: What is this line exactly?

Comment (jgkamat) [reply slides/week3/git.org:48]: this is a graph view of a git repository with no history, other than a single commit.

The hash comes first, then a `*`, then the branches, and then the commit message.

The view makes a lot more sense when you have more commits like this one:
https://github.com/RoboJackets/software-training/pull/28/files#diff-24947194ef57e8d5493a70ae16486593R135

I also generate a sample view before this slide here to prime people about the format:
https://github.com/RoboJackets/software-training/pull/28/files#diff-24947194ef57e8d5493a70ae16486593R38

Comment (kberzinch) [reply slides/week3/git.org:48]: I thought it looked something like that, but `git` on my machine doesn't format output like that...
```
kristaps@KB-XPS15:/mnt/c/Users/Kristaps/Documents/GitHub/jay-software-training$ git log --oneline --graph --all
* 28ad8c5 Slides to make pr
* 183dba8 Branching edits
*   2771727 Merge branch 'weekThreeGitSlides' of https://github.com/jgkamat/software-training into weekThreeGitSlides
|\
| * 16f86c1 Add export of git log
| * 88407a2 Add first draft of week 3 git slides
* |   f3af921 Merge pull request #27 from jgkamat/jay/vendor-reveal
|\ \
| * | f4ef4e9 Attempt to fix old artifacts staying in gh-pages
| * | 5a357df Update reveal link to point to RJ fork
|/ /
* |   3652c31 Merge pull request #21 from RoboJackets/week1fix
|\ \
| |/
|/|
| *   32f0cc9 Merge branch 'master' into week1fix
| |\
| |/
```

Comment (jgkamat) [reply slides/week3/git.org:48]: You can use `git log --oneline --graph --all --decorate` to get something similar. I should probably add that to the slides.

Comment (kberzinch) [reply slides/week3/git.org:48]: Adding the `--decorate` flag doesn't appear to change anything on my machine.

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org]: Can we just say "checked out" here?

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org]: lets you switch

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org]: Extraneous to?

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org]: Extraneous comma, again note about extraneous to?

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org]: I would rearrange these bullet points like so:
```
- Right now, the stable and the beta branch have 'forked'
- We need to bring them back together: this is called a *merge*
```

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org]: clone

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org:149]: I'm no expert but I'm not sure if merge is the right term to use here. Maybe get/send commits?

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org]: These bullets could be condensed into one:
```
- A pull request (PR) is a request for a project owner to merge a branch from your fork into their repository
```

I like the diagram but I would space out the `->` to make the labels a little easier to read.

File: slides/week3/git.org
Comment (kberzinch) [inline slides/week3/git.org]: We should figure out how we want to teach pull requests - personally I'm not a fan of every student opening a PR to this repo.

**Review by kberzinch — CHANGES_REQUESTED** [review]
I like the content but a couple things could be worded more clearly.

File: slides/week3/git.org
Comment (jgkamat) [reply slides/week3/git.org:48]: Hmm, it looks like a bunch of bugs relating to `--decorate` were fixed in git 2.4.3. Maybe you have an older version of git installed? Either way, it dosen't matter too much to this excersize if it dosen't work for others (since people can use graphical viewers like gitk, git gui, tig, etc). Hopefully people will be confident enough at this point to find something that works for them.

Comment (kberzinch) [reply slides/week3/git.org:48]:
```
kristaps@KB-XPS15:/mnt/c/Users/Kristaps/Documents/GitHub/software-training/slides$ git --version
git version 2.7.4
```
¯\\\_(ツ)\_/¯

Comment (kberzinch) [reply slides/week3/git.org:48]: I think it's fine.

File: slides/week3/git.org
Comment (jgkamat) [reply slides/week3/git.org:149]: this is still a merge (plus a fetch). You can even use the merge command to do it (with a fetch).

I want to make an explicit point here, how branches on remotes are no different from branches locally, and the same terminology and logic can be used. If remotes are treated specially it makes understanding git harder, at least in my opinion. This was one of the confusion points when I learnt git.

```
git fetch origin
git merge origin/master
```

EDIT: a push is just a merge, then a reverse fetch, or you can think of a push as the remote doing the merge/fetch from above.

Comment (kberzinch) [reply slides/week3/git.org:149]: Alright.

File: slides/week3/git.org
Comment (jgkamat) [reply slides/week3/git.org]: I'm fine with everyone opening PR's on my fork, if that works 😄

Comment (kberzinch) [reply slides/week3/git.org]: Per Slack we decided to demonstrate a pull request without everyone actually submitting one. I'd replace this slide with a simple demo title slide.

Still need to work out where that's going to happen exactly...

**Comment by jgkamat** [comment]
Putting this on hold for robocup, I agree with all the comments I didn't reply to though. If no one gets to it before me, I'll fix this up when I get back (hopefully) :smile:.

Feel free to queue more things while I'm busy though.

**Comment by joshhting** [comment]
I'll address everything here by the monday training meeting

File: slides/week3/git.org
Comment (joshhting) [inline slides/week3/git.org:39]: This image gets cut off on my screen, can we fix that somehow

Comment (kberzinch) [reply slides/week3/git.org:39]: You mean the output? I also have that issue.

**Review by kberzinch — APPROVED** [review]

File: slides/week3/git.org
Comment (jgkamat) [reply slides/week3/git.org:39]: we can just decrease the `head -n9`. I suspect that it's an issue with us overflowing reveal js again.

**Pull request merged**
